src/topp/PercolatorAdapter.cpp

```diff
@@ -1028,12 +1032,32 @@ class PercolatorAdapter :
             hit->setMetaValue("MS:1001492", pr->second.score);  // svm score
             hit->setMetaValue("MS:1001491", pr->second.qvalue);  // percolator q value
             hit->setMetaValue("MS:1001493", pr->second.posterior_error_prob);  // percolator pep
-            hit->setScore(pr->second.qvalue);
+
+            if (scoreType == "q-value")
+            {
+              hit->setScore(pr->second.qvalue);
+            }
+            else if (scoreType == "Posterior Error Probability")
+            {
+              hit->setScore(pr->second.posterior_error_prob);
+            }
+            else if (scoreType == "SVM Score")
+            {
+              hit->setScore(pr->second.score);
+            }
+
             ++cnt;
           }
           else
           {
-            hit->setScore(1.0); // set q-value to 1.0 if hit not found in results
+            if (scoreType == "q-value" || scoreType == "Posterior Error Probability")
+            {
+              hit->setScore(1.0); // set q-value or PEP to 1.0 if hit not found in results
+            }
+            else if (scoreType == "SVM Score")
+            {
+              hit->setScore(0.0); // set SVM score to 0.0 if hit not found in results
```

Comment:
Is it possible that Percolator produces scores between -1.0 and 1.0? I'm not sure if I remember correctly, but in that case we should set the SVM score to the lower boundary.

Reply:
SVM scores can be negative indeed but they are not constrained to lie in a (-1,1) interval. The absolute value of an SVM score corresponds to the distance to the SVM's margin. So, I am not really sure what to use as a lower boundary here. Just a high negative number, e.g. 1000?

Reply:
Sry, -1000 of course...

Reply:
Apparently, it happens quite a lot - at least for the test data set I used (over 500 times). Not sure why though.

Reply:
I am not sure if that is the same reason for the missing scores but I fixed it for my data in:
https://github.com/OpenMS/OpenMS/pull/3630
The reason in my case was, that terminal modifications with bracket notation e.g. PEPTIDE.[1.2345] was removed and the peptide not properly mapped to the PercolatorResult.